### DISCUSSION TITLE
Filtered realtime subscriptions

### DISCUSSION THREAD

**Discussion by HugoOlsson** [pocketbase/pocketbase#2859] [Q&A] [answered]
Hi,

I looked at the documentation and interpreted the subscriptions to be on a single document or the entire collection. Is it possible to add a filter in the subscription so that users only receive records within a condition?

Example: let’s say that I have a collection of messages from different online groups. All messages are stored in a messages collection. If a user only wants to receive updates from their group, could a filter be added to the subscription? Like “groupID = XYZ123” for the messages. 

If it is not possible, is there some other good solution to this?

Storing all the messages in a single record and subscribing only to that seems suboptimal if the number of messages becomes big. 

Thank you for the work in Pocketbase. It is an impressive project.

**Comment by HugoOlsson** [comment]
I could of course create a collection for every group, but I'm note sure if that's a good architecture.

**Comment by ganigeorgiev** [comment, marked as answer]
> Is it possible to add a filter in the subscription so that users only receive records within a condition?

Yes, but this condition can be defined only by admins via the collection List or View [API rules](https://pocketbase.io/docs/api-rules-and-filters/) _(depending on whether you've subscribe to the wildcard `*` or to a specific record)_.

Client-side level support for `?filter` and `?expand` eventually will be implemented as part of https://github.com/pocketbase/pocketbase/issues/1657 and https://github.com/pocketbase/pocketbase/issues/527.

**Reply by HugoOlsson** [reply]
Ok, thank you for the answer!
